rs/tests/networking/canister_http_correctness_test.rs

```diff
@@ -610,30 +814,1006 @@ pub fn test_http_calls_to_ic_fails(env: TestEnv) {
+            request: CanisterHttpRequestArgs {
+                url: "https://240.0.0.0".to_string(),
+                headers: BoundedHttpHeaders::new(vec![]),
+                method: HttpMethod::GET,
+                body: Some("".as_bytes().to_vec()),
+                transform: Some(TransformContext {
+                    function: TransformFunc(candid::Func {
+                        principal: get_proxy_canister_id(&env).into(),
+                        method: "transform".to_string(),
+                    }),
+                    context: vec![0, 1, 2],
+                }),
+                max_response_bytes: None,
+            },
+            cycles: 500_000_000_000,
+        },
+    ));
+
+    assert_matches!(
+        response,
+        Err(RejectResponse {
+            reject_code: RejectCode::SysTransient,
+            ..
+        })
+    );
+}
+
+/// Test that the response body returned is the same as the requested path.
+fn test_get_hello_world_call(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+    let expected_body = "hello_world";
+
+    let url = format!(
+        "https://[{}]:20443/{}/{}",
+        webserver_ipv6, "ascii", expected_body
+    );
+
+    let max_response_bytes = 666;
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers: BoundedHttpHeaders::new(vec![]),
+        method: HttpMethod::GET,
+        body: Some("".as_bytes().to_vec()),
+        transform: None,
+        max_response_bytes: Some(max_response_bytes),
+    };
+
+    let response = block_on(submit_outcall(
+        &handlers,
+        RemoteHttpRequest {
+            request: request.clone(),
+            cycles: 500_000_000_000,
+        },
+    ))
+    .expect("Request is successful.");
+
+    assert_matches!(&response, RemoteHttpResponse {body, status: 200, ..} if body == expected_body);
+    assert_http_response(&response);
+}
+
+fn test_post_call(env: TestEnv) {
+    let handlers = Handlers::new(&env);
+    let webserver_ipv6 = get_universal_vm_address(&env);
+    let expected_body = "POST";
+
+    let url = format!("https://[{}]:20443/{}", webserver_ipv6, "anything");
+    let body = Some("hello_world".as_bytes().to_vec());
+    let headers = BoundedHttpHeaders::new(vec![
+        HttpHeader {
+            name: "name1".to_string(),
+            value: "value1".to_string(),
+        },
+        HttpHeader {
+            name: "name2".to_string(),
+            value: "value2".to_string(),
+        },
+    ]);
+    let max_response_bytes = Some(666);
+
+    let request = CanisterHttpRequestArgs {
+        url,
+        headers,
+        method: HttpMethod::POST,
+        body,
+        transform: Some(TransformContext {
+            function: TransformFunc(candid::Func {
+                principal: get_proxy_canister_id(&env).into(),
+                method: "transform".to_string(),
+            }),
+            context: vec![0, 1, 2],
+        }),
```

Comment:
let's disable transform here: this test works because the transform doesn't change the body, but this assumption is fragile